docs/design/simple-rolling-update.md

```diff
@@ -0,0 +1,84 @@
+## Simple rolling update
+This is a lightweight design document for simple rolling update in ```kubectl```
+
+### Lightweight rollout
+Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
+
+```kubectl rolling-update rc foo [foo-v2] --image=myimage:v2```
+
+If the user doesn't specify a name for the 'next' controller, then the 'next' controller is renamed to
+the name of the original controller.
+
+Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
+See Recovery below
+
+If the user does specify a name for the 'next' controller, then the 'next' controller is retained with its existing name,
+and the old 'foo' controller is deleted.
+
+#### Recovery
+If a rollout fails or is terminated in the middle, it is important that the user be able to resume the roll out.
+To facilitate recovery in the case of a crash of the updating process itself, we add the following annotations to each replicaController:
+   * ```desired-replicas``` The desired number of replicas for this controller (either N or zero)
+   * ```update-partner``` A pointer to the resource that is the other half of this update (syntax ```<namespace>:<id>```)
```

Comment:
Not on the cli.  / separates resources and names right now.  I'm anti-cross namespace actions that don't involve Linux pipes

> On Apr 21, 2015, at 4:30 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In docs/design/simple-rolling-update.md:
> 
> > +```kubectl rolling-update rc foo [foo-v2] --image=myimage:v2```
> > +
> > +If the user doesn't specify a name for the 'next' controller, then the 'next' controller is renamed to
> > +the name of the original controller.
> > +
> > +Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
> > +See Recovery below
> > +
> > +If the user does specify a name for the 'next' controller, then the 'next' controller is retained with its existing name,
> > +and the old 'foo' controller is deleted.
> > +
> > +#### Recovery
> > +If a rollout fails or is terminated in the middle, it is important that the user be able to resume the roll out.
> > +To facilitate recovery in the case of a crash of the updating process itself, we add the following annotations to each replicaController:
> > +   * ```desired-replicas``` The desired number of replicas for this controller (either N or zero)
> > +   * ```update-partner``` A pointer to the resource that is the other half of this update (syntax ```<namespace>:<id>```)
> s/id/name
> 
> in a few places we used "namespace/name" syntax - prefer to stay consistent
> 
> —
> Reply to this email directly or view it on GitHub.
>